spec/models/manageiq/providers/azure_stack/cloud_manager/vcr_specs/targeted_refresh_spec.rb

```diff
@@ -0,0 +1,204 @@
+              end
+              let!(:port) do
+                FactoryBot.create(:network_port_azure_stack,
+                                  :ems_id         => ems.network_manager.id,
+                                  :ems_ref        => network_port_ref,
+                                  :name           => "dummy",
+                                  :mac_address    => "dummy",
+                                  :resource_group => resource_group)
+              end
+              let!(:security_group) do
+                FactoryBot.create(:security_group,
+                                  :ems_id         => ems.network_manager.id,
+                                  :ems_ref        => security_group_ref,
+                                  :name           => "dummy",
+                                  :resource_group => resource_group)
+              end
+
+              it "resource group is updated" do
+                test_targeted_refresh([resource_group], 'resource_group') do
+                  assert_updated(resource_group, :name => resource_group_name)
+                  assert_updated(vm, :name            => vm_name,
+                                     :raw_power_state => "PowerState/running",
+                                     :flavor          => flavor)
+                  assert_updated(network, :name => network_name)
+                  assert_updated(port, :name => port_name, :mac_address => "001DD8B700FC")
+                  assert_updated(security_group, :name => security_group_name)
+                end
+              end
+            end
+
+            context "objects are deleted from the remote server" do
+              let(:resource_group) do
+                FactoryBot.create(:resource_group_azure_stack,
+                                  :ems_id  => ems.id,
+                                  :ems_ref => "/subscriptions/#{subscription}/resourcegroups/nonexistent")
+              end
+
+              it "resource group is deleted" do
+                test_targeted_refresh([resource_group], "resource_group_deleted", :repeat => 1) do
+                  assert_deleted(resource_group)
+                end
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def test_targeted_refresh(targets, cassette, repeat: 2)
+    targets = ars_to_targets(targets)
+    repeat.times do # Run twice to verify that a second run with existing data does not change anything
+      EmsRefresh.queue_refresh(targets)
+      expect(MiqQueue.where(:method_name => 'refresh').count).to eq 1
+      refresh_job = MiqQueue.where(:method_name => 'refresh').first
+
+      vcr_with_auth("#{described_class.name.underscore}_targeted/#{api_version}-#{cassette}") do
+        status, msg, _ = refresh_job.deliver
+        expect(:status => status, :msg => msg).not_to include(:status => 'error')
+      end
+
+      ems.reload
+      yield
+    end
+  end
+
+  def ars_to_targets(targets)
```

Comment:
I can't figure out what `ars` stands for? It just sounds so much like a german "arsh" 😂 Can you help me out here?